src/MySqlConnector/Serialization/MySqlSession.cs

```diff
@@ -79,6 +86,80 @@ public MySqlSession(ConnectionPool pool, int poolGeneration)
 			ConnectionId = initialHandshake.ConnectionId;
 			AuthPluginData = initialHandshake.AuthPluginData;
 
+			if (sslMode != SslMode.None)
+			{
+				X509Certificate2 certificate;
+				try
+				{
+					certificate = new X509Certificate2(certificateFile, certificatePassword);
+				}
+				catch (CryptographicException ex)
+				{
+					if (!File.Exists(certificateFile))
+						throw new MySqlException("Cannot find SSL Certificate File", ex);
+					throw new MySqlException("Either the SSL Certificate Password is incorrect or the SSL Certificate File is invalid", ex);
+				}
+
+				Func<object, string, X509CertificateCollection, X509Certificate, string[], X509Certificate> localCertificateCb =
+					(lcbSender, lcbTargetHost, lcbLocalCertificates, lcbRemoteCertificate, lcbAcceptableIssuers) => lcbLocalCertificates[0];
+
+				Func<object, X509Certificate, X509Chain, SslPolicyErrors, bool> remoteCertificateCb =
+					(rcbSender, rcbCertificate, rcbChain, rcbPolicyErrors) =>
+				{
+					switch (rcbPolicyErrors)
+					{
+						case SslPolicyErrors.None:
+							return true;
+						case SslPolicyErrors.RemoteCertificateNameMismatch:
+							return sslMode != SslMode.VerifyFull;
+						default:
+							return sslMode == SslMode.Required;
+					}
+				};
+
+				var sslStream = new SslStream(m_tcpClient.GetStream(), false,
+					new RemoteCertificateValidationCallback(remoteCertificateCb),
+					new LocalCertificateSelectionCallback(localCertificateCb));
+				var clientCertificates = new X509CertificateCollection { certificate };
+				var sslProtocols = SslProtocols.Tls | SslProtocols.Tls11 | SslProtocols.Tls12;
```

Comment:
This is where the error on Windows is coming from. My MySQL server only speaks TLS1.1 but the handshake is trying to initiate with TLS1.2. Need to look into how to handle this properly.  Changing to `var sslProtocols = SslProtocols.Tls | SslProtocols.Tls11` on Windows works, but is not the correct solution.